buildtest/menu/show.py

```diff
@@ -1,28 +1,38 @@
 import json
-import sys
+import os
 
-from buildtest.defaults import supported_schemas
 from buildtest.buildsystem.schemas.utils import (
     load_schema,
     get_schemas_available,
     get_schema_fullpath,
+    here,
 )
 
 
 def show_schema_layout(args):
-    """Implements method ``buildtest show schema``"""
-
-    # buildtest show schema --name script
-    if args.name is None:
-        sys.exit("Please provide the name of a schema (e.g., script) to show.")
-
-    # Must be supported
-    if args.name not in supported_schemas:
-        sys.exit(
-            "%s is not a supported schema. Options include %s"
-            % (args.name, "\n".join(supported_schemas))
-        )
-
+    """This method implements command ``buildtest show schema`` which displays json schemas
+       supported by buildtest. The input ``args`` is an instance of argparse class that contains
+       user selection via command line. The method can display the global schema when
+       ``buildtest show schema --global`` is specified or any name schema via
+       ``buildtest show schema --name <NAME>``. The result is an output of the json schema on the console.
+
+       Parameters:
+
+       :param args: instance of argparse class
+       :type args: <class 'argparse.Namespace'>
+       :result: output of json schema on console
+    """
+
+    # implements ``buildtest show schema --global``. Since we can't use args.global we set this to main
+    if args.main:
+        global_schema = os.path.join(here, "global.schema.json")
+        # check if file exists
+        assert global_schema
+        recipe = load_schema(global_schema)
```

Comment:
You are doing:
```
if this evaluates to true:
    set file to be this
    load
    print
    return

set file to be this
load
print
```

and what you should do for less redundant code is

```
determine file based on input args, set file to be this
load
print
```
Hopefully that pseudo code helps :)

Reply:
Yea i see your point thanks for the hint, i will make some adjustment and get back to  you.

Reply:
Great! Design details like this are important, you will over time become a better programmer. :)